common/mb_odometry.c

```diff
@@ -33,12 +33,12 @@ void mb_initialize_odometry(mb_odometry_t* mb_odometry, float x, float y, float
 *
 *******************************************************************************/
 void mb_update_odometry(mb_odometry_t* mb_odometry, mb_state_t* mb_state){
-    float diff = mb_state->yaw_delta - mb_state->gyro_heading_delta;
+    float diff = mb_state->yaw_delta - mb_state->encoder_yaw_delta;
     
-    if (fabs(diff) > DELTA_THETA_THRESHOLD && (mb_state->left_velocity != 0 || mb_state->right_velocity != 0)) {
-        mb_state->picked_delta = mb_state->gyro_heading_delta;
-    } else {
+    if (fabs(diff) > DELTA_THETA_THRESHOLD && ( fabs(mb_state->left_velocity) >= 0.01 || fabs(mb_state->right_velocity) >= 0.01)) {
         mb_state->picked_delta = mb_state->yaw_delta;
+    } else {
+        mb_state->picked_delta = mb_state->encoder_yaw_delta;
```

Comment:
what is going on here is, when the difference between the heading from imu and encoder is larger than the user-defined threshold, and in the meantime the robot is moving, we will then ignore the heading from encoder, we trust gyro. Otherwise, we trust encoder.

Reply:
@xunfu I agree we should ignore the encoders if it gets over the threshold.... I would just put > 0 to increase readability.... >=0.01 seems a bit unnecessary

Reply:
I agree that the 0.01 is sort of confusing, the reason why I do > 0.01 is because if we do >0, even if our robot velocity is pretty small, meaning pretty closed to be still, we will still only trust the heading calculated by IMU.

Maybe we can just change it to >0 since our encoder resolution is not that fine, I guess it should be fine

Reply:
@xunfu I think we just need some reasoning behind the threshold of when to only use the imu heading vs encoder heading. How can we get to an accurate threshold? 0.05 (1/encoder resolution) could possibly be a good threshold. Or just set it > 0.

Reply:
ok let's do > 0